be/src/storage/tablet_updates.cpp

```diff
@@ -515,11 +518,13 @@ Status TabletUpdates::_rowset_commit_unlocked(int64_t version, const RowsetShare
     _next_log_id++;
     _next_rowset_id += rowsetid_add;
     EditVersionInfo* newversion = new EditVersionInfo();
+    ScopedCleanup free_guard([&]() { delete newversion; });
```

Comment:
I think you should delete the pointers in _versions in the destructor of `TabletUpdates`, not add a cleanup here.